src/definitions/device_signed.rs

```diff
@@ -25,7 +26,16 @@ pub enum DeviceAuth {
     #[serde(rename_all = "camelCase")]
     Signature { device_signature: CoseSign1 },
     #[serde(rename_all = "camelCase")]
-    Mac { device_mac: CborValue },
+    Mac { device_mac: CoseMac0 },
+}
+
+#[derive(Debug, Clone, Copy, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum DeviceAuthType {
+    #[serde(rename_all = "camelCase")]
+    Sign1,
+    #[serde(rename_all = "camelCase")]
+    Mac0,
 }
```

Comment:
Can you move this to `presentation::device`? The structs in these files are supposed to map to the data structures defined in the specification -- this is just a helper for the device session manager.

Reply:
Sure